src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,732 @@
+            }
+
+            if (!isset($schemaObject->items)) {
+                continue;
+            }
+
+            if ($schemaObject->items instanceof Schema && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponsePropertiesFromSchema($schemaObject->items, $rootType);
+            }
+
+            if ($schemaObject->items instanceof Reference && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponsePropertiesFromReference($schemaObject->items, $rootType);
+            }
+        }
+        if (current($rootType) === true) {
+            return $this->formatPropertiesToArrayOfClassInstance($this->getClassNameFromSchema($schema));
+        }
+
+        return $this->formatResponseProperties($this->getPropertiesFromSchema($schema));
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Reference $schema
+     * @param array $rootType
+     *
+     * @return array
+     */
+    protected function getReponsePropertiesFromReference(Reference $schema, array $rootType): array
+    {
+        foreach ($this->getPropertiesFromReference($schema) as $schemaObject) {
+            if ($schemaObject instanceof Schema && isset($schemaObject->properties) && !empty($schemaObject->properties)) {
+                $rootType[] = false;
+
+                return $this->getReponsePropertiesFromSchema($schemaObject, $rootType);
+            }
+            if ($schemaObject instanceof Reference && isset($schemaObject->properties) && !empty($schemaObject->properties)) {
+                $rootType[] = false;
+
+                return $this->getReponsePropertiesFromReference($schemaObject, $rootType);
+            }
+
+            if (!isset($schemaObject->items)) {
+                continue;
+            }
+
+            if ($schemaObject->items instanceof Schema && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponsePropertiesFromSchema($schemaObject->items, $rootType);
+            }
+
+            if ($schemaObject->items instanceof Reference && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponsePropertiesFromReference($schemaObject->items, $rootType);
+            }
+        }
+
+        if (current($rootType) === true) {
+            return $this->formatPropertiesToArrayOfClassInstance($this->getClassNameFromReference($schema));
+        }
+
+        return $this->formatResponseProperties($this->getPropertiesFromReference($schema));
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return array
+     */
+    protected function formatPropertiesToArrayOfClassInstance(string $className): array
```

Comment:
The method name is still not expressive enough. It's too generic and when you think of that we need more than transfer objects this should be a more concrete name.